carddav/server.go

```diff
@@ -439,3 +439,19 @@ func (b *backend) Copy(r *http.Request, dest *internal.Href, recursive, overwrit
 func (b *backend) Move(r *http.Request, dest *internal.Href, overwrite bool) (created bool, err error) {
 	panic("TODO")
 }
+
+func PreconditionError(err string) error {
+	elem := internal.NewRawXMLElement(
+		xml.Name{"urn:ietf:params:xml:ns:carddav", err},
+		[]xml.Attr{}, []internal.RawXMLValue{})
```

Comment:
`nil` can be used instead of explicit empty slices. This is shorter and avoids allocations.